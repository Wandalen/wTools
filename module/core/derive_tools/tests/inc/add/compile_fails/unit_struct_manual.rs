// T1.3: Unit struct (should not compile)
pub struct UnitStruct;

impl std::ops::Add for UnitStruct 
{
  type Output = Self;
  fn add( self, _rhs : Self ) -> Self::Output 
  {
    self.0 + _rhs.0
  }
}

fn main()
{
}
