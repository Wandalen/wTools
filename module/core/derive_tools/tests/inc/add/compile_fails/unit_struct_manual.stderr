error[E0609]: no field `0` on type `UnitStruct`
 --> tests/inc/add/compile_fails/unit_struct_manual.rs
  |
  |     self.0 + _rhs.0
  |          ^ unknown field

error[E0609]: no field `0` on type `UnitStruct`
 --> tests/inc/add/compile_fails/unit_struct_manual.rs
  |
  |     self.0 + _rhs.0
  |                   ^ unknown field
